### PR TITLE
Tidying up Godoc for Core Package

### PR DESCRIPTION
**Pull request by terencechain**
This PR marie kondo'd the Godoc comments for core package

### CODE REVIEW

File: beacon-chain/core/blocks/validity_conditions.go [hunk 1]
```diff
@@ -19,12 +16,6 @@ import (
 var log = logrus.WithField("prefix", "core/blocks")
 
 // IsValidBlock ensures that the block is compliant with the block processing validity conditions.
-// Spec:
-//  For a beacon chain block, block, to be processed by a node, the following conditions must be met:
-//  The parent block with root block.parent_root has been processed and accepted.
-//  The node has processed its state up to slot, block.slot - 1.
-//  The Ethereum 1.0 block pointed to by the state.processed_pow_receipt_root has been processed and accepted.
-//  The node's local clock time is greater than or equal to state.genesis_time + block.slot * SECONDS_PER_SLOT.
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/validity_conditions.go]: Why delete all of this?

File: beacon-chain/core/blocks/block.go [hunk 2]
```diff
@@ -37,43 +37,6 @@ func NewGenesisBlock(stateRoot []byte) *pb.BeaconBlock {
...
-// It returns an error if the requested block root is not within the BeaconState.
-// Spec pseudocode definition:
-// 	def get_block_root(state: BeaconState, slot: int) -> Hash32:
-//		"""
-//		returns the block root at a recent ``slot``.
-//		"""
-//		assert state.slot <= slot + SLOTS_PER_HISTORICAL_ROOT
-//		assert slot < state.slote
-//		return state.latest_block_roots[slot % SLOTS_PER_HISTORICAL_ROOT]
-func BlockRoot(state *pb.BeaconState, slot uint64) ([]byte, error) {
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/block.go]: Why delete these methods?

Comment (terencechain) [reply beacon-chain/core/blocks/block.go]: It's no longer used, the new method is here:
https://github.com/prysmaticlabs/prysm/blob/spec-v0.6/beacon-chain/core/helpers/block.go#L42

File: beacon-chain/core/blocks/validity_conditions.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/core/blocks/validity_conditions.go]: These comments are old, the variable names are outdated and it's no longer specified in the spec. They are moved under `ProcessBlockHeader`:
https://github.com/prysmaticlabs/prysm/blob/spec-v0.6/beacon-chain/core/blocks/block_operations.go#L84

**Pull request merged**